### PR TITLE
[NVBug 5702186] Fix awq model export for Gemma3

### PR DESCRIPTION
**Pull request by meenchen**
## What does this PR do?

**Type of change:** Bug fix  

**Overview:**  norms laers in Gemma that use (1 + weight) in forward, we will fold pre_quant_scale into the effective weight. That is to find folded w' subject to: `1 + w' = (1 + w) * s` => `w' = (1 + w) * s  -1`

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Testing

./scripts/huggingface_example.sh --model google/gemma-3-1b-it --quant int4_awq

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

* **Improvements**
  * Enhanced quantization utilities to better handle various LayerNorm variants and normalization patterns, including support for weight-offset variants and zero-centered gamma configurations.
  * Optimized pre-quantization layer normalization fusion to apply conditional weight scaling strategies based on normalization type.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: modelopt/torch/export/quant_utils.py [hunk 1]
```diff
@@ -1101,6 +1101,16 @@ def fuse_prequant_to_linear(model: torch.nn.Module, fuse_grouped_heads=False):
                     setattr(linear_pqs_from, "fused_with_prequant", True)
 
 
+def _layernorm_uses_weight_plus_one(module: torch.nn.Module) -> bool:
+    if any(
+        name in type(module).__name__
+        for name in ["LayerNorm1P", "GemmaRMSNorm", "Gemma2RMSNorm", "Gemma3RMSNorm"]
```
Comment (Edwardf0t1) [inline modelopt/torch/export/quant_utils.py:1107]: Does `LayerNorm1P` appears in Gemma3 only?

Comment (meenchen) [reply modelopt/torch/export/quant_utils.py:1107]: No, but nemotron uses it, so adding it in the change for future proof: https://github.com/huggingface/transformers/blob/24807bfcf4a21286fa2a7e728f381ddaaca7bbc7/src/transformers/models/nemotron/modeling_nemotron.py#L88

**Review by cjluo-nv — APPROVED** [review]
